src/devices/blueline.c

```diff
@@ -0,0 +1,362 @@
+            calc_crc = crc8(&offset_payload_u8[0], BLUELINE_CRC_BYTELEN, BLUELINE_CRC_POLY, BLUELINE_CRC_INIT);
+        }
+
+        // If the CRC didn't match up, ignore this row!
+        if (calc_crc != recv_crc) {
+            if ((context->searching_for_new_id) && (message_type != BLUELINE_TXID_MSG)) {
+                uint16_t id_guess = guess_blueline_id(decoder, current_row);
+                if (id_guess != 0) {
+                    if (decoder->verbose) {
+                        fprintf(stderr,"Switching to auto-detected Blueline ID %u\n", id_guess);
+                    }
+                    context->current_sensor_id = id_guess;
+                    context->searching_for_new_id = 0;
+                }
+            }
+            if (DECODE_FAIL_MIC < most_applicable_failure) {
+                most_applicable_failure = DECODE_FAIL_MIC;
+            }
+            continue;
+        }
+
+        if (message_type == BLUELINE_TXID_MSG) {
+            // TODO - Test this payload type...
+            const uint16_t received_sensor_id = ((current_row[2] << 8) | current_row[1]);
+            /* clang-format off */
+            data = data_make(
+                    "model", "", DATA_STRING, BLUELINE_MODEL,
+                    "id",    "", DATA_INT, received_sensor_id,
+                    "mic",   "", DATA_STRING, "CRC",
+                    NULL);
+            /* clang-format on */
+            decoder_output_data(decoder, data);
+            payloads_decoded++;
+            if (context->searching_for_new_id) {
+                if (decoder->verbose) {
+                    fprintf(stderr,"Switching to received Blueline ID %u\n", received_sensor_id);
+                }
+                context->current_sensor_id = received_sensor_id;
+                context->searching_for_new_id = 0;
+            }
+        } else if (message_type == BLUELINE_POWER_MSG) {
+            const uint16_t ms_per_pulse = offset_payload_u16;
+            /* clang-format off */
+            data = data_make(
+                    "model", "", DATA_STRING, BLUELINE_MODEL,
+                    "id",    "", DATA_INT, context->current_sensor_id,
+                    "gap",   "", DATA_INT,    ms_per_pulse,
+                    "mic",   "", DATA_STRING, "CRC",
+                    NULL);
+            /* clang-format on */
+            decoder_output_data(decoder, data);
+            payloads_decoded++;
+        } else if (message_type == BLUELINE_TEMPERATURE_MSG) {
+            // TODO - Add in proper temperature scaling to C or F
+            // TODO - Confirm battery flag is working properly
+
+            // These seem to be the best estimates from Powermon433,
+            // but they don't line up with what my official LCD display shows.
+            //
+            // I'm leaving the display raw while I figure this out better...
+            //
+            // A: deg_f = 0.823 * recvd_temp - 28.63
+            // B: deg_c = 0.457 * recvd_temp - 33.68
+
```

Comment:
We need to figure out the temperature encoding.

Reply:
The one from Powerman433 is close enough.  I'm just being nit-picky perfectionist and trying to figure out why it's sometimes a degree or two off from the official LCD display.  I'll put it back in for now and continue experimenting on my own.